packages/cli-kit/src/public/node/http.ts

```diff
@@ -73,3 +75,51 @@ ${sanitizedHeadersOutput((options?.headers ?? {}) as {[header: string]: string})
   outputDebug(`Request to ${url.toString()} completed with status ${response.status} in ${Math.round(t1 - t0)} ms`)
   return response
 }
+
+/**
+ * Download a file from a URL to a local path.
+ *
+ * @param url - The URL to download from.
+ * @param to - The local path to download to.
+ * @param redirect - The number of redirects that have been followed.
+ * @returns - A promise that resolves with the local path.
+ */
+export function downloadFile(url: string, to: string, redirect = 0): Promise<string> {
+  outputDebug(redirect ? `Redirecting to ${url}` : `Downloading ${url} to ${to}`)
+
+  return new Promise<string>((resolve, reject) => {
+    if (!fileExistsSync(dirname(to))) {
+      mkdirSync(dirname(to))
+    }
+
+    let done = true
+    const file = createFileWriteStream(to)
+
+    file.on('finish', () => {
+      if (done) {
+        file.close()
+        resolve(to)
+      }
+    })
+
+    file.on('error', (err) => {
+      unlinkFileSync(to)
+      reject(err)
+    })
+
+    nodeFetch(url)
+      .then((res) => {
+        if (res.status === 302 && res.headers.get('location') !== null) {
+          const redirection = res.headers.get('location')!
+          done = false
+          file.close()
+          resolve(downloadFile(redirection, to, redirect + 1))
+        }
```

Comment:
I think we can use `nodeFetch(url, redirect: 'follow')` and forget about this logic.

Reply:
ah, good catch